### PR TITLE
feat: added claim screen

### PR DESCRIPTION
**Pull request by alt-a1r**
Added Claim screen and ClaimButton component. PR is draft because it should be checked after fixes regarding useGetContract hook

About #321

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by sirpy** [comment]
@johnsmith-gooddollar all components should be in nativebase in the new goodweb3-mono ui package

**Comment by johnsmith-gooddollar** [comment]
@alt-a1r ^^^^

File: src/components/SideBar.tsx [hunk 1]
```diff
@@ -318,7 +318,7 @@ export default function SideBar({ mobile, closeSidebar }: { mobile?: boolean, cl
                             </defs>
                         </svg>
                     </div>
-                      <div className="flex flex-col details">                  
+                      <div className="flex flex-col details">
```
Comment (johnsmith-gooddollar) [inline src/components/SideBar.tsx:321]: Should be Box according nativebase docs: https://docs.nativebase.io/box

File: src/pages/gd/Claim/ClaimButton/styled.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import styled from 'styled-components'
+import { ButtonPrimary } from "../../../../components/ButtonLegacy";
+
+export const StyledButton = styled(ButtonPrimary)`
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/ClaimButton/styled.ts:4]: - all base/common styles for buttons etc should be defined by extendTheme https://docs.nativebase.io/customizing-components
- custom styles should be object (like in GD app) https://github.com/GeekyAnts/NativeBase/issues/4935#issuecomment-1144847331

File: src/pages/gd/Claim/styled.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import styled from 'styled-components'
+import {Layout} from 'components/gd/sushi'
+
+export const PageContent = styled.div`
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/styled.ts:4]: migrate to nativebase (see comments above)

File: src/pages/gd/Claim/index.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { t } from '@lingui/macro'
+import ClaimButton from './ClaimButton'
+import { useLingui } from '@lingui/react'
+import { PageContent, StyledLayout } from "./styled";
+
+const Claim = () => {
+	const { i18n } = useLingui()
+
+	return (
+		<StyledLayout classes="md:mt-24 xl:mt-0 sh:mt-30">
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/index.tsx:12]: use NativeBase's layouts: read https://docs.nativebase.io/box and sections below

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

**Comment by sirpy** [comment]
@alt-a1r @roma-hladilka @johnsmith-gooddollar 
most new components should be in the new good-design package

**Comment by alt-a1r** [comment]
@sirpy Now buttons components (ClaimButton in particular) are imported from @good-design (I haven't updated PR yet because it's not merged in the @good-design)
Do we need to move any other components there? And which components need to be moved there at all (for future PR's)

File: src/theme/nativeTheme.tsx [hunk 5]
```diff
@@ -1,29 +1,33 @@
...
-	},
-})
-
-export function NativeBaseThemeProvider({ children }: { children: React.ReactNode }) {
-	return <NativeBaseProvider theme={nativeTheme}>{children}</NativeBaseProvider>
-}
+// export function NativeBaseThemeProvider({ children }: { children: React.ReactNode }) {
+// 	return <NativeBaseProvider theme={nativeTheme}>{children}</NativeBaseProvider>
+// }
+//TODO: move logic to good-design, no native-base logic here.
```
Comment (L03TJ3) [inline src/theme/nativeTheme.tsx:29]: @alt-a1r consider this comment

File: src/pages/gd/Claim/index.tsx [hunk 6]
```diff
@@ -1,52 +1,25 @@
...
-				{i18n._(t`Claim`)}
-			</Text>
-			<View style={styles.content}>
-				{/*<ClaimButton />*/}
-			</View>
-		</View>
-	)
+    const { i18n } = useLingui()
+    return (
+        <NativeBaseProvider theme={theme}>
```
Comment (L03TJ3) [inline src/pages/gd/Claim/index.tsx:11]: @alt-a1r any styles should be defined either in the theme, or specifically styled on the component returned

File: src/pages/gd/Claim/index.tsx [hunk 7]
```diff
@@ -3,30 +3,47 @@ import { t } from '@lingui/macro'
...
-        {i18n._(t`Claim`)}
-      </Title>
-      <View style={styles.content}>
-        <ClaimButton  firstName={'Test'} method={'redirect'}/>
-      </View>
-    </Layout>
-  )
+    const { i18n } = useLingui()
+    const { account } = useActiveWeb3React()
+    return (
```
Comment (johnsmith-gooddollar) [inline src/pages/gd/Claim/index.tsx:13]:
```suggestion

    return (
```
pls add empty lines betweek code blocks, this increases readability

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

**Comment by johnsmith-gooddollar** [comment]
pls add empty lines betweek code blocks, this increases readability

**Pull request closed without merging**